write.go

```diff
@@ -192,106 +189,30 @@ func (out *OutputDir) Cleanup(secrets map[string]Secret) error {
 	return nil
 }
 
-// FileInfo returns the filesystem properties atomicWrite wrote
-type FileInfo struct {
-	Mode os.FileMode
-	UID  uint32
-	GID  uint32
-}
-
-// GetFileInfo from an open file
-func GetFileInfo(file *os.File) (*FileInfo, error) {
-	stat, err := file.Stat()
-	if err != nil {
-		return nil, fmt.Errorf("failed to stat after writing: %v", err)
-	}
-	filemode := stat.Mode()
-	uid := stat.Sys().(*syscall.Stat_t).Uid
-	gid := stat.Sys().(*syscall.Stat_t).Gid
-
-	return &FileInfo{filemode, uid, gid}, nil
-}
-
-// atomicWrite creates a temporary file, sets perms, writes content, and renames it to filename
-// This sequence ensures the following:
-// 1. Nobody can open the file before we set owner/permissions properly
-// 2. Nobody observes a partially-overwritten secret file.
-// Since keysync is intended to write to tmpfs, this function doesn't do the necessary fsyncs if it
-// were persisting content to disk.
+// Write puts a Secret into OutputDir
 func (out *OutputDir) Write(secret *Secret) (*secretState, error) {
```

Comment:
This function stays the same (and all its tests still pass), just the core file logic is factored out into output.WriteFileAtomically